src/metabase/models/table.clj

```diff
@@ -139,7 +187,7 @@
         :active          true
         :table_id        [:in table-ids]
         :visibility_type [:not= "retired"]
-        {:order-by [[:position :asc] [:name :asc]]}))
+        {:order-by [[:name :asc]]}))
```

Comment:
This is not ideal. Currently `with-fields` is only used in transforms where we don't care about field order, so I don't think it's worth the hoop jumping this would require to be both efficient and respect `:field_order`.